DIPs/DIPNNNN.md

```diff
@@ -156,17 +160,67 @@ libraries but language currently lack a way to express such semantics.
 
 ### Proposal
 
-Introduce new symbol kind concept into compiler, *future symbol*. This term does
-not refer to any language syntax and is intended to represent new concept in
-compiler internals. It has to have special treatment during semantic analysis:
+Introduce new symbol kind concept into compiler, *future symbol*. In terms of
+DMD compiler internals it can be defined simply as additional boolean field
+of `DSymbol` indicating if symbol is *future* or not.
+
+To make examples more readable, imaginary `@future` attribute will be used to
+mark such symbols. However introducing such attribute (or any other) is NOT
+part of the proposal and serves only illustration purpose.
+
+Symbol marked as *future* has to have special treatment during semantic analysis:
+
+1. Whenever compiler has to do a symbol lookup, if has to check if *future
+   symbol* is present in the result. If present, deprecation message has to be
+   printed and everything else should continue as if it wasn't present.
+
+   ```D
+   import moda; // provides foo
+   import modb; // provides @future foo
+
+   foo(); // Deprecation: upcoming adddition of modb.foo will result in a symbol
+          // clash, please disambugate
+   ```
+
+2. Whenevr compiler hasto check for presence of a symbol, if there is a match
+   which is *future symbol*, deprecation message has to be printed and
+   everything else should continue as it wasn't present.
+
+   ```D
+   class A
+   {
+       void foo() @future;
+   }
+
+   class B
+   {
+       void foo(); // Deprecation: upcoming addition of A.foo will result in a
+                   // symbol clash, please adjust
+   }
+   ```
+
+3. References to *future symbol* within the module it is declared in should not
+   result in any deprecation messages because the module can be adjusted in the
+   same time symbol is actually added.
+
+   ```D
+   void foo (long a);
+   void foo (int a) @future;
+
+   foo(42); // No point in printing deprecation here as it will keep compiling
+            // even when @future is removed
+   ```
```

Comment:
I'd use `string` and `Object` overloads, called with `null`, to make the example a bit more realistic.